lib/daru/io/io.rb

```diff
@@ -202,23 +200,16 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
-          .tap { |c| yield c if block_given? }
+          .parse(open(path), opts)
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
-          ::CSV
-          .open(path, 'rb', opts)
-          .tap { |c| yield c if block_given? }
```

Comment:
Yes, this does make sense. So what I was doing in the previous commit is what should be done, right? Differentiating between a URL and a local csv should be the way to go like this? -
```
def from_csv_hash_with_headers(path,opts) 
      url = URI.parse(path)
      if %w(http https).include?(url.scheme)
         ::CSV
           .parse(open(path), opts)
           .by_col.map { |col_name, values| [col_name, values] }.to_h
      else
         ::CSV
           .read(path, 'rb',opts)
           .tap { |c| yield c if block_given? }
           .by_col.map { |col_name, values| [col_name, values] }.to_h
      end
end
```

And similarly for the function  ```from_csv_hash```

I will write the test for passing the block too along with this. :+1: